Mathlib/CategoryTheory/Category/Basic.lean

```diff
@@ -180,9 +180,15 @@ initialize_simps_projections Category (-Hom)
 /-- postcompose an equation between morphisms by another morphism -/
 theorem eq_whisker {f g : X ⟶ Y} (w : f = g) (h : Y ⟶ Z) : f ≫ h = g ≫ h := by rw [w]
 
+theorem eq_whisker_assoc {W : C} {f g : X ⟶ Y} {h i : Y ⟶ Z} (w : f ≫ h = g ≫ i) (j : Z ⟶ W) :
```

Comment:
This looks like it shouldn't be its own lemma, and should be made as some reassoc where needed?

Reply:
`reassoc` will not produce a lemma of this form, it would produce one where `f = g` implies `f ≫ h ≫ j = g ≫ h ≫ j` for any `j`, which is not really useful. Maybe I should change the name?

Reply:
`reassoc_of% w` doesn’t work?

Reply:
Right, I think this situation is exactly what `reassoc_of` is for; Matthew's suggestion is precisely what I had in mind

Reply:
Ohh right, yeah that should probably work. However, then the next problem is that `reassoc` imports this file, so I can't use it here. But maybe I should just prove this directly in the proof that requires it below?

Reply:
Ah, that is indeed a convincing problem! I think proving it directly is a better option - this lemma isn't going to be used outside this file but since you need something like it below, probably you can get away with that. If this turns out to be prohibitively longer, however, you could alternatively make it private.